matadi/_templates.py

```diff
@@ -88,19 +94,21 @@ def __init__(self, material):
         self.x = [self.material.x[0], p, J]
         self.W = Material(self.x, self._fun)
         self.function = self.W.function
-        self.gradient = self.W.gradient
         self.hessian = self.W.hessian
         self.gradient_vector_product = self.W.gradient_vector_product
         self.hessian_vector_product = self.W.hessian_vector_product
 
     def _fun(self, x):
-        F, p, J = x
+        F, p, J = x[:3]
         F = horzcat(vertcat(x[0], zeros(1, 2)), zeros(3, 1))
         F[2, 2] = 1  # fixed thickness ratio `h / H = 1`
         detF = det(F)
         Fmod = (J / detF) ** (1 / 3) * F
         return self.material.fun(Fmod, **self.material.kwargs) + p * (detF - J)
 
+    def gradient(self, x, *args, **kwargs):
+        return [*self.W.gradient(x, *args, **kwargs), None]
+
```

Comment:
No change of `x` necessary here. Just as for hyperelastic materials, return unused state variables.